charts/flux2/templates/source-controller.yaml

```diff
@@ -109,8 +109,12 @@ spec:
       {{- if .Values.sourceController.priorityClassName }}
       priorityClassName: {{ .Values.sourceController.priorityClassName | quote }}
       {{- end }}
+      {{- if .Values.sourceController.podSecurityContext }}
+      securityContext: {{ toYaml .Values.sourceController.podSecurityContext | nindent 8 }}
+      {{- else }}
       securityContext:
         fsGroup: 1337
```

Comment:
Let’s move these defaults in values.yaml please, so we don’t need the if/else everywhere

Reply:
Thanks for taking a look @stefanprodan. I've thought about that, but I think `podSecurityContext` should work the same way as `securityContext` does. And there we already have these if/else blocks.

I've considered changing both `securityContext` and `podSecurityContext` to have the default in `values.yaml` but that would be a breaking change. With the current if/else logic, running `helm install ... --set kustomizeController.securityContext.runAsNonRoot=false` will result in a Deployment with:

```yaml
securityContext:
  runAsNonRoot: false
```

While, with having the default in `values.yaml`, the same command results in:

```yaml
securityContext:
  allowPrivilegeEscalation: false
  capabilities:
    drop:
    - ALL
  readOnlyRootFilesystem: true
  runAsNonRoot: false
  seccompProfile:
    type: RuntimeDefault
```

Reply:
Oh Ok good point. thanks for looking into it.

Reply:
No problem. Is there anything else you think we could improve on?